common/types.go

```diff
@@ -58,6 +61,15 @@ func (h *Hash) UnmarshalJSON(input []byte) error {
 	if length >= 2 && input[0] == '"' && input[length-1] == '"' {
 		input = input[1 : length-1]
 	}
+	// strip "0x" for length check
+	if len(input) > 1 && string(input[:2]) == "0x" {
```

Comment:
Yeah I know, but nonce of the methods in `common` handle `0X` as of now, so it's weird to add it here but leave all the various other locations as `0x` only.